Paintroid/src/main/java/org/catrobat/paintroid/command/implementation/MergeLayersCommand.kt

```diff
@@ -28,18 +29,31 @@ class MergeLayersCommand(position: Int, mergeWith: Int) : Command {
     var position = position; private set
     var mergeWith = mergeWith; private set
 
+    companion object {
+        private val TAG = MergeLayersCommand::class.java.simpleName
+    }
+
     override fun run(canvas: Canvas, layerModel: LayerContracts.Model) {
         val sourceLayer = layerModel.getLayerAt(position)
         val destinationLayer = layerModel.getLayerAt(mergeWith)
-        val destinationBitmap = destinationLayer.bitmap
-        destinationBitmap ?: return
-        val copyBitmap = destinationBitmap.copy(destinationBitmap.config, true)
-        val copyCanvas = Canvas(copyBitmap)
-        copyCanvas.drawBitmap(sourceLayer.bitmap ?: return, 0f, 0f, null)
-        destinationLayer.bitmap = copyBitmap
-        layerModel.removeLayerAt(position)
-        if (sourceLayer === layerModel.currentLayer) {
-            layerModel.currentLayer = destinationLayer
+        var success = false
+        if (sourceLayer != null && destinationLayer != null) {
+            val destinationBitmap = destinationLayer.bitmap
+            destinationBitmap ?: return
+            val copyBitmap = destinationBitmap.copy(destinationBitmap.config, true)
+            val copyCanvas = Canvas(copyBitmap)
+            copyCanvas.drawBitmap(sourceLayer.bitmap ?: return, 0f, 0f, null)
+            if (layerModel.removeLayerAt(position)) {
+                destinationLayer.bitmap = copyBitmap
+                if (sourceLayer === layerModel.currentLayer) {
```

Comment:
you can simply write `==` instead of `===`

Reply:
`===` was introduced by a previous commit and i would leave it there. It should be the same reference in this case and a referential equality compare is faster than a structural. I don't know the specific reason but it should not harm and would leave it as it is 😃

Reply:
yeah I know it was introduced in a previous commit, I was just doubtful whether these two values would always point to the same object when they're structurally equal 🤔. If you think they'll be, then it's fine :)

Reply:
Since we both have our doubts, lets change it to `==`! :)

Reply:
thanks, it would be better ig :)